RxSwiftWorkshop/Maps/Model/GeolocatorSpec.swift

```diff
@@ -34,47 +33,89 @@ class GeolocatorSpec: QuickSpec {
                 sut = nil
             }
 
-            it("should pass geolocation request to its SystemGeolocator") {
-                guard case let (disposeBag?, systemGeolocator?) = unwrap(disposeBag, systemGeolocator) else { return }
+            condition("always") {
                 let address = "Valencia Airport"
-                sut?.geolocate(address: address).subscribe().disposed(by: disposeBag)
-                systemGeolocator.verifyCall(
-                    withIdentifier: FakeSystemGeolocator.Identifiers.geocodeAddressString, arguments: [address]
-                )
+                setup {
+                    sut?.geolocate(address: address).subscribe().disposed(by: disposeBag!)
+                }
+                it("should pass geolocation request to its SystemGeolocator") {
+                    systemGeolocator!.verifyCall(
+                        withIdentifier: FakeSystemGeolocator.Identifiers.geocodeAddressString, arguments: [address]
+                    )
+                }
             }
 
-            it("should return geolocation from its SystemGeolocator") {
-                guard case let (sut?, disposeBag?, systemGeolocator?) = unwrap(sut, disposeBag, systemGeolocator) else { return }
+            condition("when SystemGeolocator succeeds in finding location") {
                 let location = CLLocation(latitude: 123.123, longitude: 132.132)
-                systemGeolocator.fixedLocation = location
                 var result: CLLocation?
-                sut.geolocate(address: "Valencia Airport").subscribe(onNext: { result = $0 }).disposed(by: disposeBag)
-                expect(result?.coordinate).to(equal(location.coordinate))
+                setup {
+                    systemGeolocator!.fixedLocation = location
+                    sut!.geolocate(address: "Valencia Airport")
```

Comment:
I'm still struggling with this one. `SystemGeolocator` has no reactive API nor anything to do with reactiveness at all, and therefore it's strange for `FakeSystemGeolocator` to have information wrapped into observables instead of fixtures. The asynchronicity, however, might take place (because of `completionBlock` in `SystemGeolocator`). The way we deal with that is the implementation of `FakeSystemGeolocator`. Whether `FakeSystemGeolocator` takes an observable or a fixture, it's in its implementation details to pass the location in an synchronous way. I don't really see how the scheduler changes that. Could you elaborate further?